lib/datasource/nuget/v3.ts

```diff
@@ -29,97 +39,114 @@ export async function getQueryUrl(url: string): Promise<string | null> {
   }
 
   try {
-    // TODO: fix types
-    const servicesIndexRaw = await http.getJson<any>(url);
-    const searchQueryService = servicesIndexRaw.body.resources.find(
-      (resource) => resource['@type']?.startsWith(resourceType)
+    const servicesIndexRaw = await http.getJson<ServicesIndexRaw>(url);
+    const services = servicesIndexRaw.body.resources.find((resource) =>
+      resource['@type']?.startsWith(resourceType)
     );
-    const searchQueryServiceId = searchQueryService['@id'];
+    const serviceId = services['@id'];
 
     const cacheMinutes = 60;
-    await packageCache.set(
-      cacheNamespace,
-      cacheKey,
-      searchQueryServiceId,
-      cacheMinutes
-    );
-    return searchQueryServiceId;
+    await packageCache.set(cacheNamespace, cacheKey, serviceId, cacheMinutes);
+    return serviceId;
   } catch (e) {
     logger.debug(
       { e },
-      `nuget registry failure: can't get SearchQueryService form ${url}`
+      `nuget registry failure: can't get ${resourceType} form ${url}`
     );
     return null;
   }
 }
 
+interface CatalogEntry {
+  version: string;
+  published?: string;
+  projectUrl?: string;
+}
+
+interface CatalogPage {
+  '@id': string;
+  items: {
+    catalogEntry: CatalogEntry;
+  }[];
+}
+
+interface PackageRegistration {
+  items: CatalogPage[];
+}
+
+async function getCatalogEntry(
+  catalogPage: CatalogPage
+): Promise<CatalogEntry[]> {
+  let items = catalogPage.items;
+  if (!items) {
+    const url = catalogPage['@id'];
+    const catalogPageFull = await http.getJson<CatalogPage>(url);
+    items = catalogPageFull.body.items;
+  }
+  return items.map(({ catalogEntry }) => catalogEntry);
+}
+
 export async function getReleases(
   registryUrl: string,
   feedUrl: string,
   pkgName: string
 ): Promise<ReleaseResult | null> {
-  let queryUrl = `${feedUrl}?q=${pkgName}`;
-  if (registryUrl.toLowerCase() === defaultNugetFeed.toLowerCase()) {
-    queryUrl = queryUrl.replace('q=', 'q=PackageId:');
-    queryUrl += '&semVerLevel=2.0.0&prerelease=true';
+  const url = `${feedUrl.replace(/\/*$/, '')}/${pkgName}/index.json`;
+  const packageRegistration = await http.getJson<PackageRegistration>(url);
+  const catalogPages = packageRegistration.body.items || [];
+  const catalogPagesQueue = catalogPages.map((page) => (): Promise<
+    CatalogEntry[]
+  > => getCatalogEntry(page));
```

Comment:
Why do you fetch the CatalogEntry again? The required metadata is aready included. see
https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource#sample-response

Reply:
Seems like it paginates for releases with more than 64 items. So `getCatalogEntry` have to make additional requests if needed.
Example:
https://api.nuget.org/v3/registration3/nunit/index.json (items included)
https://api.nuget.org/v3/registration5-semver1/nlog/index.json (paginated)

Reply:
Ah, I see, this is because of `registration5-semver1` endpoint

Reply:
I was wrong, here is the example of paginated response for the latest `RegistrationsBaseUrl`:
https://api.nuget.org/v3/registration5-gz-semver2/nlog/index.json

Reply:
ok, so only fetch page if items is missing 👍